src/NServiceBus.Core/Transports/Msmq/MessagePump.cs

```diff
@@ -146,36 +146,41 @@ public void Start(PushRuntimeSettings limitations)
 
                     await concurrencyLimiter.WaitAsync(cancellationToken).ConfigureAwait(false);
 
-                    var tokenSource = new CancellationTokenSource();
-
                     var receiveTask = Task.Run(async () =>
                     {
-                        try
-                        {
-                            await receiveStrategy.ReceiveMessage(inputQueue, errorQueue, tokenSource, pipeline).ConfigureAwait(false);
-                            receiveCircuitBreaker.Success();
-                        }
-                        catch (MessageQueueException ex)
+                        using (var tokenSource = new CancellationTokenSource())
```

Comment:
Since the implementations are the ones using the token source if they need it (some don't), it should be their job to create it. But I'm ok with leaving this out of scope of this PR.